crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1318,6 +1348,13 @@ where
         }
 
         if !state.retryable {
+            if state.tx_finalized_with_different_user_sig
```

Comment:
The only time `tx_finalized_with_different_user_sig` is true is if `check_if_error_indicates_tx_finalized_with_different_user_sig` returns true in `record_non_quorum_effects_maybe`. Seems like we don't need this property in state as we also check that same method here.

Reply:
> The only time tx_finalized_with_different_user_sig is true is if check_if_error_indicates_tx_finalized_with_different_user_sig returns true in record_non_quorum_effects_maybe

This is right. However it's possible that in `record_non_quorum_effects_maybe`, `check_if_error_indicates_tx_finalized_with_different_user_sig` is not called hence `tx_finalized_with_different_user_sig` is not updated.

Reply:
we can think `tx_finalized_with_different_user_sig` as a cached computation result